endpoints/openrtb2/auction_test.go

```diff
@@ -120,196 +191,178 @@ func TestExplicitUserId(t *testing.T) {
 }
 
-// TestBadRequests makes sure we return 400s on requests with bad Native requests.
-func TestBadNativeRequests(t *testing.T) {
-	tests := &getResponseFromDirectory{
-		dir:           "sample-requests/invalid-native",
-		payloadGetter: buildNativeRequest,
-		messageGetter: nilReturner,
-		expectedCode:  http.StatusBadRequest,
-		aliased:       false,
+func doRequest(t *testing.T, test testCaseData) (int, string) {
+	disabledBidders := map[string]string{
+		"indexExchange": "Bidder \"indexExchange\" has been deprecated and is no longer available. Please use bidder \"ix\" and note that the bidder params have changed.",
 	}
-	tests.assert(t)
+	bidderMap := exchange.DisableBidders(getBidderInfos(test.Config.adaptersConfig, openrtb_ext.BidderList()), disabledBidders)
+
+	// NewMetrics() will create a new go_metrics MetricsEngine, bypassing the need for a crafted configuration set to support it.
+	// As a side effect this gives us some coverage of the go_metrics piece of the metrics engine.
+	theMetrics := pbsmetrics.NewMetrics(metrics.NewRegistry(), openrtb_ext.BidderList(), config.DisabledMetrics{})
+
+	endpoint, _ := NewEndpoint(
+		&nobidExchange{},
+		newParamsValidator(t),
+		&mockStoredReqFetcher{},
+		empty_fetcher.EmptyFetcher{},
+		&config.Configuration{
+			MaxRequestSize:     maxSize,
+			BlacklistedApps:    test.Config.BlacklistedAppArr,
+			BlacklistedAppMap:  test.Config.blacklistedAppMap,
+			BlacklistedAccts:   test.Config.BlacklistedAccountArr,
+			BlacklistedAcctMap: test.Config.blacklistedAccountMap,
+			AccountRequired:    test.Config.AccountReq,
+		},
+		theMetrics,
+		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
+		disabledBidders,
+		[]byte(test.Config.AliasJSON),
+		bidderMap,
+	)
+
+	request := httptest.NewRequest("POST", "/openrtb2/auction", bytes.NewReader(test.BidRequest))
+	recorder := httptest.NewRecorder()
+	endpoint(recorder, request, nil) //Request comes from the unmarshalled mockBidRequest
+	return recorder.Code, recorder.Body.String()
 }
 
-// TestAliasedRequests makes sure we handle (default) aliased bidders properly
-func TestAliasedRequests(t *testing.T) {
-	tests := &getResponseFromDirectory{
-		dir:           "sample-requests/aliased",
-		payloadGetter: noop,
-		messageGetter: nilReturner,
-		expectedCode:  http.StatusOK,
-		aliased:       true,
+func jsonParseTestFile(t *testing.T, fileData []byte, testFile string) *testCaseData {
+	t.Helper()
+
+	parsedTestData := &testCaseData{}
+	var err error
+
+	// Get testCaseData values
+	parsedTestData.BidRequest, _, _, err = jsonparser.Get(fileData, "mockBidRequest")
+	assert.NoError(t, err, "Error jsonparsing root.mockBidRequest from file %s. Desc: %v.", testFile, err)
+
+	parsedReturnCode, err := jsonparser.GetInt(fileData, "expectedReturnCode")
+	assert.NoError(t, err, "Error jsonparsing root.code from file %s. Desc: %v.", testFile, err)
+
+	parsedTestData.ExpectedReturnCode = int(parsedReturnCode)
+
+	if parsedTestData.ExpectedReturnCode != 200 {
```

Comment:
This doesn't seem like the right place to make this check. Consider loading both the ExpectedErrorMessage and ExpectedBidResponse.

Reply:
Moved the check and refactored to load both `ExpectedBidResponse` and  `ExpectedErrorMessage` but I still assert error based on the return code. Let me know if that's the approach you had in mind.